File: ToDoList.Server/Models/SubToDoModel.cs
using static ToDoList.Server.Controllers.ToDoListController;

namespace ToDoList.Server.Models
{
    public class SubToDoModel
    {
        public int Id { get; set; }
        public int ParentId {  get; set; }
        public string Task { get; set; }
        public DateTime Deadline { get; set; }
        public bool IsCompleted { get; set; }
        public string? MoreDetails { get; set; }
        public bool ShowDetails { get; set; }
    }
}

File: todolist.client/src/app/models/subTodoModel.ts
export class SubTodoModel {
  constructor() { }
  public task: string = "";
  public deadline: Date = new Date();
  public moreDetails: string = "";
  public id: number = 0;
  public parentId: number = 0;
  public isCompleted: boolean = false;
  public showDetails?: boolean = false;
}
